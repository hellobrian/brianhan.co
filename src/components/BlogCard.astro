---
import ReadMore from "../components/ReadMore.astro";
export interface Props {
  href: string;
  title: string;
  summary: string;
}

const { href, title, summary } = Astro.props;
---

<article>
  <h3>
    <a href={href}>{title}</a>
  </h3>
  <p>
    {summary}
  </p>
  <ReadMore href={href} />
</article>

<style>
  :root {
    --color--summary--dark: #999999;
    --color--summary--light: #474747;
  }

  article {
    display: flex;
    flex-direction: column;
    gap: 8px;
    max-width: 670px;
  }

  h3 > a {
    color: var(--foreground);
    text-decoration: none;
  }

  h3 > a:hover {
    text-decoration: underline;
    text-underline-offset: 2px;
    text-decoration-thickness: 1.2px;
  }

  p {
    font-weight: 500;
  }

  :global(.darkmode p) {
    color: var(--color--summary--dark);
  }

  :global(p) {
    color: var(--color--summary--light);
  }
</style>
