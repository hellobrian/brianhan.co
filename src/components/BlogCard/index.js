import React from 'react';
import PropTypes from 'prop-types';
import StyledLink from 'src/components/StyledLink';

const BlogCard = ({ title, subtitle, excerpt, date, to, href }) => {
  return (
    <div
      css={`
        padding-left: 2rem;
        padding-right: 2rem;
        margin-bottom: 4rem;
        min-width: 400px;

        &:last-child {
          margin-bottom: 0;
        }

        @media screen and (min-width: 400px) {
          margin-bottom: 1rem;
        }

        @media screen and (min-width: 800px) {
          box-shadow: var(--card-shadow);
          padding-bottom: 1rem;
          padding-left: 2rem;
          padding-right: 2rem;
          padding-top: 3rem;
        }
      `}>
      <h2
        css={`
          line-height: 1.5;
          padding-bottom: 0;
          margin-bottom: 0;

          @media (max-width: 800px) {
            font-size: 1.25rem;
          }
        `}>
        <StyledLink to={to} href={href}>
          {title}
        </StyledLink>
      </h2>
      <p
        css={`
          font-family: 'Karla', sans-serif;
          margin-top: 5px;
          margin-bottom: 5px;
        `}>
        {date}
      </p>
      <p>{subtitle || excerpt}</p>
    </div>
  );
};

BlogCard.propTypes = {
  date: PropTypes.string.isRequired,
  excerpt: PropTypes.string,
  subtitle: PropTypes.string,
  title: PropTypes.string.isRequired,
  to: PropTypes.string,
  href: PropTypes.string,
};

BlogCard.defaultProps = {
  to: '',
  href: '',
};

export default BlogCard;
