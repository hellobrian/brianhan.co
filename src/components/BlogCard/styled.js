import styled from 'styled-components';

export const Root = styled.div`
  margin-bottom: 4rem;

  &:last-child {
    margin-bottom: 0;
  }
`;

export const Title = styled.h2`
  line-height: 1.5;
  padding-bottom: 0;

  @media (max-width: 800px) {
    font-size: 1.25rem;
  }
`;

export const Date = styled.p`
  font-family: 'Karla', sans-serif;
`;
