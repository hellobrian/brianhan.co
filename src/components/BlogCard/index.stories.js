import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text } from '@storybook/addon-knobs';
import BlogCard from './index';

const stories = storiesOf('BlogCard', module);
stories.addDecorator(withKnobs);
stories.add('default', () => (
  <BlogCard
    title={text('title', 'Adopting Prettier Early at Work')}
    fromNow={text('fromNow', '5 days ago')}
    excerpt={text('excerpt', excerpt)}
  />
));

const excerpt = `These days you can easily create an online resume or 
portfolio using just some HTML templates. With a little CSS and JS you can style your text, 
add interactive charts or display your previous works in a professional way.`;
