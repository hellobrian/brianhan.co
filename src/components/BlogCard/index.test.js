import React from 'react';
import { render } from '@testing-library/react';
import BlogCard from './index';

describe('BlogCard', () => {
  it('renders correctly', () => {
    const { container } = render(
      <BlogCard
        title="title"
        subtitle="subtitle"
        excerpt="excerpt"
        date="date"
        to="/"
      />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
