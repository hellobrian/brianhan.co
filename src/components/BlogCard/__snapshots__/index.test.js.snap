// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogCard renders correctly 1`] = `
.c3 {
  background-repeat: no-repeat;
  background-image: var(--gradient--link);
  background-size: 100% 3px;
  background-position: 0 100%;
}

.c2 {
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: currentColor;
}

.c0 {
  padding-left: 2rem;
  padding-right: 2rem;
  margin-bottom: 4rem;
  min-width: 400px;
}

.c0:last-child {
  margin-bottom: 0;
}

.c1 {
  line-height: 1.5;
  padding-bottom: 0;
  margin-bottom: 0;
}

.c4 {
  font-family: 'Karla',sans-serif;
  margin-top: 5px;
  margin-bottom: 5px;
}

@media screen and (min-width:400px) {
  .c0 {
    margin-bottom: 1rem;
  }
}

@media screen and (min-width:800px) {
  .c0 {
    box-shadow: var(--card-shadow);
    padding-bottom: 1rem;
    padding-left: 2rem;
    padding-right: 2rem;
    padding-top: 3rem;
  }
}

@media (max-width:800px) {
  .c1 {
    font-size: 1.25rem;
  }
}

<div
  class="c0"
>
  <h2
    class="c1"
  >
    <a
      class="c2"
      href="/"
    >
      <span
        class="c3"
      >
        title
      </span>
    </a>
  </h2>
  <p
    class="c4"
  >
    date
  </p>
  <p>
    subtitle
  </p>
</div>
`;
